flake.nix

```diff
@@ -35,6 +35,7 @@
               cabal-install
               doctest
               haskell-ci
+              haskell-language-server
```

Comment:
Does this work? Haskell Language Server requires the exact same GHC for its own building and for its usage. The GHCs were pulled from `pkgs.haskell.packages.${version}`, but hls is always from `pkgs.haskellPackages` which is the default version I think?

Reply:
I think I've fixed this?

Reply:
Probably. Have you tested if HLS in non-default shells work? (I'm ok without the test though, we can delegate that to anyone who first uses HLS in other shells)